fink_science/cats/processor.py

```diff
@@ -128,66 +121,81 @@ def predict_nn(
 
     filter_dict = {'u': 1, 'g': 2, 'r': 3, 'i': 4, 'z': 5, 'Y': 6}
 
-    bands = []
-    lcs = []
+    mjd = []
+    filters = []
     meta = []
-    frac = 10**(- (31.4 - 27.5) / 2.5)
 
     for i, mjds in enumerate(midpointTai):
 
         if len(mjds) > 0:
-            bands.append(np.array(
+            filters.append(np.array(
                 [filter_dict[f] for f in filterName.values[i]]
             ).astype(np.int16))
-            lc = np.concatenate(
-                [
-                    mjds[:, None],
-                    frac * psFlux.values[i][:, None],
-                    frac * psFluxErr.values[i][:, None]
-                ], axis=-1
-            )
 
-            if not np.isnan(mwebv.values[i]):
+            mjd.append(mjds - mjds[0])
 
-                lcs.append(normalize_lc(lc).astype(np.float32))
-
-                meta.append([
-                    mwebv.values[i], z_final.values[i],
-                    z_final_err.values[i], hostgal_zphot.values[i],
-                    hostgal_zphot_err.values[i]
-                ])
-
-    X = {
-        'meta': np.array(meta),
-        'band': tf.keras.preprocessing.sequence.pad_sequences(bands,
-                                                              maxlen=243,
-                                                              dtype='int32'),
-        'lc': tf.keras.preprocessing.sequence.pad_sequences(lcs,
-                                                            maxlen=243,
-                                                            dtype='float32'),
-    }
+            if not np.isnan(mwebv.values[i]):
 
-    for i, x in enumerate(X['meta'][:, 3]):
-        if x < 0:
-            X['meta'][i, 1:] = -1
-        else:
-            X['meta'][i, 1:] = x
+                meta.append([mwebv.values[i],
+                             hostgal_zphot.values[i],
+                             hostgal_zphot_err.values[i],
+                             z_final.values[i],
+                             z_final_err.values[i]])
+
+    flux = psFlux.apply(lambda x: norm_column(x))
+    mjd = midpointTai.apply(lambda x: norm_column(x))
+    error = psFluxErr.apply(lambda x: norm_column(x))
+
+    flux = keras.utils.pad_sequences(flux,
```

Comment:
The CI is failing with:

```
    AttributeError: module 'keras.api._v2.keras.utils' has no attribute 'pad_sequences'
```

which I think comes from a mismatch with the keras version. Which version of tensorflow are you using?

Reply:
Hi Julien, we're using tensorflow==2.9.2

Reply:
Thanks -- I checked with this version, and there is no problem! So you can keep it as is.